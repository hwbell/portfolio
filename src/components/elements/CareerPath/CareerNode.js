import React from "react";

import { toolsImages } from "../../../assets/toolsImages";
import { Button, Modal, ModalHeader, ModalBody, ModalFooter } from "reactstrap";

class CareerNode extends React.Component {
  constructor(props) {
    super(props);

    this.state = {};
  }

  renderTools() {
    let imageId;
    return this.props.tools.map((tool, i) => {
      imageId = `${this.props.job.split(" ").join("")}-tool${i}`;

      return (
        <div id={imageId} className="tool-image col" key={i} style={styles.toolHolder}>
          <a
            target="_blank"
            href={`https://www.google.com/search?q=${tool}`}
            alt={`link to ${tool}`}
          >
            <img
              // onMouseOver={() => {
              //   this.togglePopover(tool);
              // }}
              // onMouseLeave={() => {
              //   this.togglePopover();
              // }}
              style={styles.toolImage}
              src={toolsImages[tool]}
              alt="no image"
            />
          </a>
        </div>
      );
    });
  }

  render() {
    console.log(this.props.logo);
    return (
      <div className="career-node" style={styles.main}>
        <img
          src={this.props.logo}
          alt={`image for ${this.props.job}`}
          style={styles.logo}
        />

        {/* <div className="left-all-row row">{this.renderTools()}</div> */}
      </div>
    );
  }
}

const styles = {
  toolHolder: {
    maxWidth: '50px',
    backgroundColor: 'white'
  },
  toolImage: {
    height: "35px",
    width: "35px"
  },
  logo: {
    borderRadius: '10px',
    width: "100px",
  },
};

export default CareerNode;
