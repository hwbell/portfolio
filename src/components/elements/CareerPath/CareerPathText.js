const CareerPathText = [
  {
    year: "2010",
    job: "Brandeis University",
    tools: [ "matlab", "dna", "microscope"],
    logo: require("../../../assets/icons/brandeis.png")
  },
  {
    year: "2017",
    job: "University of colorado",
    tools: [ "matlab", "dna", "microscope"],
    logo: require("../../../assets/icons/brandeis.png")
  },
  {
    year: "2019",
    job: "Granicus Inc.",
    tools: [ "html5", "css3", "javascript", "php", "jira"],
    logo: require("../../../assets/icons/brandeis.png")
  },
  {
    year: "2020",
    job: "The Next Step",
    tools: [ "html5", "css3", "javascript", "reactjs", "nodejs"],
    logo: require("../../../assets/icons/brandeis.png")
  }
]

export {
  CareerPathText
}