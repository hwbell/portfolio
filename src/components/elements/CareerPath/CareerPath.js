import React from "react";

// data
import { CareerPathText } from "./CareerPathText";

// elements
import CareerNode from "./CareerNode";

class CareerPath extends React.Component {
  constructor(props) {
    super(props);

    this.state = {};
  }

  renderNodes() {
    return (
      <div className="career-nodes-container center-all-row row">
        {CareerPathText.map((obj, i) => {
          return (
            <CareerNode {...obj} index={i} key={i}/>
          )
        })}
      </div>
    );
  }

  render() {
    return (
      <div id="career-path-container" style={styles.main}>
        {CareerPathText && this.renderNodes()}
      </div>
    );
  }
}

const styles = {
  main: {},
};

export default CareerPath;
