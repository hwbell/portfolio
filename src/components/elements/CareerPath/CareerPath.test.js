import React from 'react';
import ReactDOM from 'react-dom';
import CareerPath from './CareerPath';
import CareerNode from './CareerNode';

import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';

// react router context
// import { MemoryRouter } from 'react-router-dom';

describe('CareerPath Component', () => {

  let wrapper;
  let someProps;
  beforeEach(() => {
    someProps = {

    }

    wrapper = mount(
      // only need the MemoryRouter if there are links to routes of the react-router in the component that we need to test
      // <MemoryRouter>
      <CareerPath {...someProps} />
      // </MemoryRouter>
    );

  })

  // tests
  it('renders without crashing', async () => {
    wrapper.update();
  });

  it('renders correctly', async () => {
    const tree = await renderer
      .create(<CareerPath {...someProps} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('contains the correct elements for different data', () => {
    console.log(wrapper)

    expect(wrapper.find('#career-path-container').length).toEqual(1);
    expect(wrapper.find('.career-nodes-container').length).toEqual(1);

  })

  // it('should render all buttons with info from provided props.buttons', () => {
  //   expect(wrapper.find('Link').at(0).props().to).toBe(someProps.buttons[0].link)
  //   expect(wrapper.find('Link').at(1).props().to).toBe(someProps.buttons[1].link)
  //   expect(wrapper.find('Link').at(2).props().to).toBe(someProps.buttons[2].link)

  // })
})