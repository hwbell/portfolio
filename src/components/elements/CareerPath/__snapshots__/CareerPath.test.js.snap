// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CareerPath Component renders correctly 1`] = `
<div
  id="career-path-container"
  style={Object {}}
>
  <div
    className="career-nodes-container"
  >
    <div
      className="career-node"
      style={
        Object {
          "image": Object {
            "width": "100px",
          },
          "toolImage": Object {
            "height": "35px",
          },
        }
      }
    >
      <img
        alt="image for Brandeis University"
      />
      <div
        className="tool-image"
        id="BrandeisUniversity-tool0"
      >
        <a
          alt="link to matlab"
          href="https://www.google.com/search?q=matlab"
          target="_blank"
        >
          <img
            alt="no image"
            onMouseLeave={[Function]}
            onMouseOver={[Function]}
            src="matlab.jpeg"
          />
        </a>
      </div>
      <div
        className="tool-image"
        id="BrandeisUniversity-tool1"
      >
        <a
          alt="link to dna"
          href="https://www.google.com/search?q=dna"
          target="_blank"
        >
          <img
            alt="no image"
            onMouseLeave={[Function]}
            onMouseOver={[Function]}
          />
        </a>
      </div>
      <div
        className="tool-image"
        id="BrandeisUniversity-tool2"
      >
        <a
          alt="link to microscope"
          href="https://www.google.com/search?q=microscope"
          target="_blank"
        >
          <img
            alt="no image"
            onMouseLeave={[Function]}
            onMouseOver={[Function]}
          />
        </a>
      </div>
    </div>
    <div
      className="career-node"
      style={
        Object {
          "image": Object {
            "width": "100px",
          },
          "toolImage": Object {
            "height": "35px",
          },
        }
      }
    >
      <img
        alt="image for University of colorado"
      />
      <div
        className="tool-image"
        id="Universityofcolorado-tool0"
      >
        <a
          alt="link to matlab"
          href="https://www.google.com/search?q=matlab"
          target="_blank"
        >
          <img
            alt="no image"
            onMouseLeave={[Function]}
            onMouseOver={[Function]}
            src="matlab.jpeg"
          />
        </a>
      </div>
      <div
        className="tool-image"
        id="Universityofcolorado-tool1"
      >
        <a
          alt="link to dna"
          href="https://www.google.com/search?q=dna"
          target="_blank"
        >
          <img
            alt="no image"
            onMouseLeave={[Function]}
            onMouseOver={[Function]}
          />
        </a>
      </div>
      <div
        className="tool-image"
        id="Universityofcolorado-tool2"
      >
        <a
          alt="link to microscope"
          href="https://www.google.com/search?q=microscope"
          target="_blank"
        >
          <img
            alt="no image"
            onMouseLeave={[Function]}
            onMouseOver={[Function]}
          />
        </a>
      </div>
    </div>
    <div
      className="career-node"
      style={
        Object {
          "image": Object {
            "width": "100px",
          },
          "toolImage": Object {
            "height": "35px",
          },
        }
      }
    >
      <img
        alt="image for Granicus Inc."
      />
      <div
        className="tool-image"
        id="GranicusInc.-tool0"
      >
        <a
          alt="link to html5"
          href="https://www.google.com/search?q=html5"
          target="_blank"
        >
          <img
            alt="no image"
            onMouseLeave={[Function]}
            onMouseOver={[Function]}
            src="html5.png"
          />
        </a>
      </div>
      <div
        className="tool-image"
        id="GranicusInc.-tool1"
      >
        <a
          alt="link to css3"
          href="https://www.google.com/search?q=css3"
          target="_blank"
        >
          <img
            alt="no image"
            onMouseLeave={[Function]}
            onMouseOver={[Function]}
            src="css3.png"
          />
        </a>
      </div>
      <div
        className="tool-image"
        id="GranicusInc.-tool2"
      >
        <a
          alt="link to javasript"
          href="https://www.google.com/search?q=javasript"
          target="_blank"
        >
          <img
            alt="no image"
            onMouseLeave={[Function]}
            onMouseOver={[Function]}
          />
        </a>
      </div>
      <div
        className="tool-image"
        id="GranicusInc.-tool3"
      >
        <a
          alt="link to php"
          href="https://www.google.com/search?q=php"
          target="_blank"
        >
          <img
            alt="no image"
            onMouseLeave={[Function]}
            onMouseOver={[Function]}
            src="php.png"
          />
        </a>
      </div>
      <div
        className="tool-image"
        id="GranicusInc.-tool4"
      >
        <a
          alt="link to jira"
          href="https://www.google.com/search?q=jira"
          target="_blank"
        >
          <img
            alt="no image"
            onMouseLeave={[Function]}
            onMouseOver={[Function]}
            src="jira.png"
          />
        </a>
      </div>
    </div>
    <div
      className="career-node"
      style={
        Object {
          "image": Object {
            "width": "100px",
          },
          "toolImage": Object {
            "height": "35px",
          },
        }
      }
    >
      <img
        alt="image for The Next Step"
      />
      <div
        className="tool-image"
        id="TheNextStep-tool0"
      >
        <a
          alt="link to html5"
          href="https://www.google.com/search?q=html5"
          target="_blank"
        >
          <img
            alt="no image"
            onMouseLeave={[Function]}
            onMouseOver={[Function]}
            src="html5.png"
          />
        </a>
      </div>
      <div
        className="tool-image"
        id="TheNextStep-tool1"
      >
        <a
          alt="link to css3"
          href="https://www.google.com/search?q=css3"
          target="_blank"
        >
          <img
            alt="no image"
            onMouseLeave={[Function]}
            onMouseOver={[Function]}
            src="css3.png"
          />
        </a>
      </div>
      <div
        className="tool-image"
        id="TheNextStep-tool2"
      >
        <a
          alt="link to javascript"
          href="https://www.google.com/search?q=javascript"
          target="_blank"
        >
          <img
            alt="no image"
            onMouseLeave={[Function]}
            onMouseOver={[Function]}
            src="javascript.png"
          />
        </a>
      </div>
      <div
        className="tool-image"
        id="TheNextStep-tool3"
      >
        <a
          alt="link to reactjs"
          href="https://www.google.com/search?q=reactjs"
          target="_blank"
        >
          <img
            alt="no image"
            onMouseLeave={[Function]}
            onMouseOver={[Function]}
            src="react.png"
          />
        </a>
      </div>
      <div
        className="tool-image"
        id="TheNextStep-tool4"
      >
        <a
          alt="link to nodejs"
          href="https://www.google.com/search?q=nodejs"
          target="_blank"
        >
          <img
            alt="no image"
            onMouseLeave={[Function]}
            onMouseOver={[Function]}
            src="nodejs.png"
          />
        </a>
      </div>
    </div>
  </div>
</div>
`;
